src/main/java/mate/academy/bookstore/repository/book/BookSpecificationProviderManager.java

```diff
@@ -0,0 +1,23 @@
+package mate.academy.bookstore.repository.book;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.repository.SpecificationProvider;
+import mate.academy.bookstore.repository.SpecificationProviderManager;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class BookSpecificationProviderManager implements SpecificationProviderManager<Book> {
+    private final List<SpecificationProvider<Book>> bookSpecificationProviders;
+
+    @Override
+    public SpecificationProvider<Book> getSpecificationProvider(String key) {
+        return bookSpecificationProviders.stream()
+                .filter(p -> p.getKey().equals(key))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException(

```

Comment:
change p to provider for clarity, replace exception with more precise and suitable NoSuchElementException